### DISCUSSION TITLE
polybar showing underneath default ubuntu top bar

### DISCUSSION THREAD

**Discussion by kaykhan** [polybar/polybar#2793] [Q&A]
Hi, I am on Ubuntu 22.04 and i have compiled and installed the latest release of polybar. When i run `polybar` https://github.com/polybar/polybar/wiki. it seems like the existing ubuntu top bar is covering the poly bar. Im not sure if this is how its suppose to be... ( fyi i dont use any custom window manager)

![image](https://user-images.githubusercontent.com/5202527/185805133-a2c1570f-1af1-449e-8384-17f766f2a91c.png)

**polybar**
```
notice: Parsing config file: /etc/polybar/config.ini
notice: Listening for IPC messages (PID: 7508)
notice: Loading module 'xworkspaces' of type 'internal/xworkspaces'
notice: Loading module 'xwindow' of type 'internal/xwindow'
notice: Loading module 'filesystem' of type 'internal/fs'
notice: Loading module 'pulseaudio' of type 'internal/pulseaudio'
notice: pulseaudio: using default sink alsa_output.pci-0000_00_05.0.analog-stereo
notice: Loading module 'xkeyboard' of type 'internal/xkeyboard'
notice: Loading module 'memory' of type 'internal/memory'
notice: Loading module 'cpu' of type 'internal/cpu'
notice: Loading module 'wlan' of type 'internal/network'
error: Disabling module "wlan" (reason: No interface found for type 'wireless')
notice: Loading module 'eth' of type 'internal/network'
notice: module/eth: Discovered wired interface enp0s3
notice: Loading module 'date' of type 'internal/date'
notice: Loaded 9 modules
notice: Loaded font "monospace" (name=DejaVu Sans Mono, offset=2, file=/usr/share/fonts/truetype/dejavu/DejaVuSansMono.ttf)
warn: background_manager: Failed to get root pixmap, default to black (is there a wallpaper?)
warn: background_manager: Failed to get root pixmap, default to black (is there a wallpaper?)
warn: background_manager: Failed to get root pixmap, default to black (is there a wallpaper?)
warn: background_manager: Failed to get root pixmap, default to black (is there a wallpaper?)
warn: background_manager: Failed to get root pixmap, default to black (is there a wallpaper?)

```

**Comment by patrick96** [comment]
In the default config, polybar takes no action to position itself above or below other windows (with `wm-restack = generic` you can tell it to put itself below all other windows, but that will give you the same result). Whether polybar is below or above the gnome bar seems to be entirely controlled by your desktop environment. I would just remove the gnome bar, move polybar down a bit using `offset-y` (may not work in gnome), or put polybar at the bottom of the screen (`bottom = true`).

**Reply by kaykhan** [reply]
It sounds like youre suggesting to me to remove the gnome bar in favour of polybar

Do you have any reccomendations on how to remove the gnome bar?  (scanning through the settings in ubuntu, there is no option that allows me to remove it)

**Reply by patrick96** [reply]
> It sounds like youre suggesting to me to remove the gnome bar in favour of polybar

That's just one of the solutions. Having polybar on top of the gnome bar isn't that desirable either. Also, polybar isn't a replacement for gnome bar, they have vastly different functionalities. If you need gnome bar, you should probably move polybar somewhere else.

I think there is a gnome plugin somewhere that can do that, last time I checked there was no built-in way to remove it.

**Reply by kaykhan** [reply]
Ok sounds like there might some functionality in gnome top bar that i can't get with polybar... i don't really wnat to move it else where, would idealy like to implement at the top and add existing gnome bar funcitonality to it .

**Comment by grmrgecko** [comment]
What I do is I add Transparent Top Bar extension to make the gnome topbar 100% transparent. I then configure height and whatnot to match the gnome topbar in polybar, and set a transparent background for polybar to get a nice effect.
